examples-gallery/plot_sliding_block.py

```diff
@@ -0,0 +1,333 @@
+# %%
+""""
+Sliding a Block on a Road
+=========================
+A block, modeled as a particle is sliding on a road to cross a
+hill. The block is subject to gravity and speed dependent
+friction.
+Gravity points in the negative Y direction.
+A force tangential to the road is applied to the block.
+Three objective functions to me minimized may be selected:
+
+- selektion = 0: time to reach the end point is minimized
+- selektion = 1: time to reach the end point and the energy are minimized.
+- selektion = 2: energy consumed is minimized.
+
+**Constants**
+
+- m: mass of the block [kg]
+- g: acceleration due to gravity [m/s**2]
+- reibung: coefficient of friction [N/(m*s)]
+- a, b: paramenters determining the shape of the road.
+
+**States**
+
+- x: position of the block [m]
+- ux: velocity of the block [m/s]
+
+**Specifieds**
+
+- F: force applied to the block [N]
+
+"""
+import sympy.physics.mechanics as me
+from collections import OrderedDict
+import numpy as np
+import sympy as sm
+from opty.direct_collocation import Problem
+import matplotlib.pyplot as plt
+import matplotlib as mp
+from matplotlib.animation import FuncAnimation
+from copy import deepcopy
+# %%
+# define the road
+def strasse(x, a, b):
+    return a * x**2 * sm.exp((b - x))
+
+# %%
+# set up Kane's EOMs
+N = me.ReferenceFrame('N')
+O = me.Point('O')
+O.set_vel(N, 0)
+t = me.dynamicsymbols._t
+
+P0 = me.Point('P0')
+x = me.dynamicsymbols('x')
+ux = me.dynamicsymbols('u_x')
+F = me.dynamicsymbols('F')
+
+m, g, reibung = sm.symbols('m, g, reibung')
+a, b = sm.symbols('a b')
+
+P0.set_pos(O, x * N.x + strasse(x, a, b) * N.y)
+P0.set_vel(N, ux * N.x + strasse(x, a, b).diff(x)*ux * N.y)
+BODY = [me.Particle('P0', P0, m)]
+
+# The control force and the friction are acting in the direction of
+# the tangent at the street at the point whre the particle is.
```

Comment:
These show up as code comments, Make them "cells" so that they display as rendered text.

Reply:
meaning like:
# %%
# The control force.....
# the tangent...
?

Reply:
Cells are delimited by the `# %%`

Reply:
That's what I meant. Typo again! :-(